### PR TITLE
Plugins: Add API for creating pluginv2 proto client

### PR DESCRIPTION
**Pull request by wbrowne**
**What is this feature?**

Adds an API to construct a client for the `pluginv2` plugin protobuf

**Why do we need this feature?**

It's useful in cases where we don't need to use the plugin SDK `backend` package which acts as an API wrapper around the underlying protobuf.

**Who is this feature for?**

Plugins Platform

Please check that:
- [X] It works as expected from a user's perspective.
- [X] If this is a pre-GA feature, it is behind a feature toggle.
- [X] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/plugins/backendplugin/grpcplugin/client_proto.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+type ProtoClientOpts struct {
+	PluginID       string
+	ExecutablePath string
+	ExecutableArgs []string
+	Env            []string
+	Logger         log.Logger
+}
+
+func NewProtoClient(opts ProtoClientOpts) (ProtoClient, error) {
+	p := &grpcPlugin{
```
Comment (marefr) [inline pkg/plugins/backendplugin/grpcplugin/client_proto.go:46]: Could you use `newPlugin` here to reuse code?

**Review by marefr — APPROVED** [review]
LGTM with a comment/question

File: pkg/plugins/backendplugin/grpcplugin/client_proto.go (hunk 1 above)
Comment (wbrowne) [reply pkg/plugins/backendplugin/grpcplugin/client_proto.go:46]: Do you mean the `newPlugin` func that returns a `backendplugin.Plugin` interface?

Comment (marefr) [reply pkg/plugins/backendplugin/grpcplugin/client_proto.go]: Yes that was my idea. But we discussed offline and feel free to merge to move this forward

**Review by xnyo — APPROVED** [review]
LGTM!

**Pull request merged**